projects/hello_firmware/src/main.c

```diff
@@ -0,0 +1,48 @@
+#include <stdint.h>  // for integer types
+
+#include "interrupt.h"  // interrupts are required for soft timers
+#include "log.h"
+#include "soft_timer.h"  // Software timers for scheduling future events.
+#include "wait.h"        // for wait function
+
+#define COUNTER_A_TIMER 500
```

Comment:
Please be a bit more descriptive with naming here, e.g. `COUNTER_A_PERIOD_MS` where the `_MS` suffix is to denote milliseconds. We usually use a units suffix for `#define`s that denote real values.